drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableObjectStore.java

```diff
@@ -141,4 +154,52 @@ public void safepoint() {
             storage.flush();
         }
     }
+
+    private void updateReferencedObjects(StoredObject object){
+        List<Field> referencedObjects = getReferencedObjects(object.getObject());
+        if (referencedObjects.size()>0) {
+
+        }
+    }
+
+    private StoredObject setReferencedObjects(StoredObject object){
+        List<Field> referencedObjects = getReferencedObjects(object.getObject());
+        if (referencedObjects.size()>0) {
+            // for each referenced object in sObject
+            //  lookup in storage, find the object of reference, get its fact handle id
+            //      save this association in the StoredObject
+            referencedObjects.forEach(field -> {
+                field.setAccessible(true);
+                Object fieldObject = null;
+                try {
+                    fieldObject = field.get(object.getObject());
+                } catch (IllegalAccessException e) {
+                    e.printStackTrace();
+                }
+                Long objectKey = fromObjectToFactHandleId(fieldObject);
+                if (objectKey!=null){
+                ((SerializableStoredObject) object).addReferencedObject(field.getName(), objectKey);}
+            });
+        }
+        return object;
+    }
+
+    private Long fromObjectToFactHandleId(Object object){
+        for (Long key : this.storage.keySet()){
+            if (( (SerializableStoredObject) storage.get(key)).getObject().hashCode()==object.hashCode()){
```

Comment:
```suggestion
            if (((SerializableStoredObject) storage.get(key)).getObject() == object) {
```
Isn't this what you want to do?

The same `hashCode` value doesn't always mean that the objects are equal.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()

Reply:
Yes, that's true, thank you Toshiya, I will try the "==" operator.